packages/terra-theme-provider/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ThemeProviderThemes = {
+  CONSUMER: 'cerner-consumer-theme',
+  MOCK: 'cerner-mock-theme',
+};
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by `<ThemeProvider />`
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `cerner-consumer-theme`
+   */
+  themeName: PropTypes.string,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => (
+  <div {...customProps} className={themeName}>{children}</div>
+);
+
+ThemeProvider.propTypes = propTypes;
+ThemeProvider.Opts = {};
+ThemeProvider.Opts.Themes = ThemeProviderThemes;
```

Comment:
Yeah, the user can pass in whatever string they want and that will be applied as the class name. Using the `ThemeProvider.Opts.Themes` object, users have access to the list of names of all the first class themes.